cub/cub/detail/choose_offset.cuh

```diff
@@ -56,6 +56,36 @@ struct ChooseOffsetT
                                          unsigned long long>::type;
 };
 
+template <typename T,typename = void>
+struct is_iterator : std::false_type {};
+template <typename T>
+struct is_iterator<T,std::void_t<typename std::iterator_traits<T>::iterator_category>> : std::true_type {};
+template <typename T>
+inline constexpr bool is_iterator_v = is_iterator<T>::value;
+
+// checks if input type is a valid offset iterator and assigns iterator's value_type to member ::type
+template <typename OffsetIterT>
+struct offset_iter_value {
+	static_assert(
+		is_iterator_v<OffsetIterT>,
```

Comment:
This needs to use the trait in C++11 we have a convenience macro for that in libcu++
```suggestion
		_LIBCUDACXX_TRAIT(is_iterator, OffsetIterT),
```